lib/plugins/aws/lib/updateStack.js

```diff
@@ -33,43 +34,72 @@ module.exports = {
       stackTags = Object.assign(stackTags, this.serverless.service.provider.stackTags);
     }
 
-    const params = {
+    const createChangeSetParams = {
       StackName: stackName,
-      OnFailure: 'DELETE',
+      ChangeSetName: changeSetName,
+      ChangeSetType: 'CREATE',
       Capabilities: ['CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM'],
       Parameters: [],
       TemplateURL: templateUrl,
       Tags: Object.keys(stackTags).map((key) => ({ Key: key, Value: stackTags[key] })),
     };
 
+    const executeChangeSetParams = {
+      StackName: stackName,
+      ChangeSetName: changeSetName,
+    };
+
     if (
       this.serverless.service.provider.compiledCloudFormationTemplate &&
       this.serverless.service.provider.compiledCloudFormationTemplate.Transform
     ) {
-      params.Capabilities.push('CAPABILITY_AUTO_EXPAND');
+      createChangeSetParams.Capabilities.push('CAPABILITY_AUTO_EXPAND');
     }
 
     const customDeploymentRole = this.provider.getCustomDeploymentRole();
     if (customDeploymentRole) {
-      params.RoleARN = customDeploymentRole;
+      createChangeSetParams.RoleARN = customDeploymentRole;
     }
 
     if (this.serverless.service.provider.notificationArns) {
-      params.NotificationARNs = this.serverless.service.provider.notificationArns;
+      createChangeSetParams.NotificationARNs = this.serverless.service.provider.notificationArns;
     }
 
     if (this.serverless.service.provider.stackParameters) {
-      params.Parameters = this.serverless.service.provider.stackParameters;
+      createChangeSetParams.Parameters = this.serverless.service.provider.stackParameters;
     }
 
     if (this.serverless.service.provider.disableRollback) {
-      delete params.OnFailure;
-      params.DisableRollback = this.serverless.service.provider.disableRollback;
+      executeChangeSetParams.DisableRollback = this.serverless.service.provider.disableRollback;
     }
 
-    return this.provider
-      .request('CloudFormation', 'createStack', params)
-      .then((cfData) => this.monitorStack('create', cfData));
+    // Create new change set
+    log.info('Creating new change set');
+    const changeSet = await this.provider.request(
```

Comment:
Shouldn't we remove changeset created with previous deployment first?

Reply:
This is a fallback for create so the stack doesn't exist yet - in such case there's no previous changeset to be removed

Reply:
Thanks for clarification 👍